server/data/data-library/src/main/java/org/apache/james/rrt/lib/RecipientRewriteTableManagement.java

```diff
@@ -37,136 +38,136 @@
 public class RecipientRewriteTableManagement extends StandardMBean implements RecipientRewriteTableManagementMBean {
 
     private final RecipientRewriteTable rrt;
-    
+
     @Inject
     protected RecipientRewriteTableManagement(RecipientRewriteTable rrt) throws NotCompliantMBeanException {
         super(RecipientRewriteTableManagementMBean.class);
         this.rrt = rrt;
     }
 
     @Override
-    public void addRegexMapping(String user, String domain, String regex) throws Exception {
+    public void addRegexMapping(String user, String domain, String regex) {
         try {
             rrt.addRegexMapping(user, Domain.of(domain), regex);
         } catch (RecipientRewriteTableException e) {
-            throw new Exception(e.getMessage());
+            throw Throwables.propagate(e);
```

Comment:
I don't understand why it's wrapped into an Exception given that Exception is certainly a super type of RecipientRewriteTableException

Reply:
Good catch. Sorry I missed this, obviously...

Reply:
Thanks for the explaination for `Throwables.propagate`. I bet we need to add a task to migrate away from it.